cmd/maestro/agent/cmd.go

```diff
@@ -47,53 +58,98 @@ func NewAgentCommand() *cobra.Command {
 	fs := cmd.PersistentFlags()
 	fs.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
 	fs.AddGoFlagSet(flag.CommandLine)
-	commonOptions.CommoOpts.AddFlags(fs)
-	addFlags(fs)
+	agentOptions.AddFlags(fs)
+
 	utilruntime.Must(features.SpokeMutableFeatureGate.Add(ocmfeature.DefaultSpokeWorkFeatureGates))
 	utilruntime.Must(features.SpokeMutableFeatureGate.Set(fmt.Sprintf("%s=true", ocmfeature.RawFeedbackJsonString)))
 
 	return cmd
 }
 
-func runAgent(cmd *cobra.Command, args []string) {
-	ctx, cancel := context.WithCancel(context.Background())
-
-	stopCh := make(chan os.Signal, 1)
-	signal.Notify(stopCh, syscall.SIGINT, syscall.SIGTERM)
-	go func() {
-		defer cancel()
-		<-stopCh
-	}()
+type AgentOptions struct {
+	CommonOptions  *options.AgentOptions
+	WorkOptions    *spoke.WorkloadAgentOptions
+	KubeConfigFile string
+	Namespace      string
+}
 
+func NewAgentOptions() *AgentOptions {
+	workOptions := spoke.NewWorkloadAgentOptions()
+	// use 1M as the default limit for state feedback
+	workOptions.MaxJSONRawLength = 1024 * 1024
 	// use mqtt as the default driver
-	agentOption.MaxJSONRawLength = maxJSONRawLength
-	cfg := spoke.NewWorkAgentConfig(commonOptions, agentOption)
-	cmdConfig := commonOptions.CommoOpts.
-		NewControllerCommandConfig("maestro-agent", version.Get(), cfg.RunWorkloadAgent)
-	cmdConfig.DisableLeaderElection = true
-
-	if err := cmdConfig.StartController(ctx); err != nil {
-		glog.Fatalf("error running command: %v", err)
+	workOptions.WorkloadSourceDriver = "mqtt"
+	// use manifest as the default codec
+	workOptions.CloudEventsClientCodecs = []string{"manifest"}
```

Comment:
Could you update to support `manifest` and `manifestbundle` by default?

Reply:
I'm not sure, we should enable both codecs at the same time, for maestro, it should only handle the manifest, right?

Reply:
Let us handle it in separate issue https://github.com/openshift-online/maestro/issues/96